lightning/src/offers/merkle.rs

```diff
@@ -167,16 +185,46 @@ impl<'a> TlvStream<'a> {
 	fn skip_signatures(self) -> core::iter::Filter<TlvStream<'a>, fn(&TlvRecord) -> bool> {
 		self.filter(|record| !SIGNATURE_TYPES.contains(&record.r#type))
 	}
+
+	pub(crate) fn intersection(self, rhs: Self) -> Vec<u8> {
```

Comment:
I could be wrong but I doubt this will ever be much of a performance bottleneck (and if it does it'll be pretty easy to fix in the future) so I decided to remove it